client/boot/app.js

```diff
@@ -42,7 +42,7 @@ const boot = currentUser => {
 		setupMiddlewares( currentUser, reduxStore );
 		invoke( project, 'setupMiddlewares', currentUser, reduxStore );
 		detectHistoryNavigation.start();
-		page.start();
+		page.start( { decodeURLComponents: false } );
```

Comment:
This is a pretty big change, we should check that it doesn't break other parts of calypso.
Basically this is page.js trying to be too smart and auto decoding the querystring. ([Related issue](https://github.com/visionmedia/page.js/issues/216))
This causes problems when one of the parameter is an url which also has query params since those will be decoded and considered valid query params when parsing `context.querystring` to construct `context.query`